### PR TITLE
359 create a test suite which verifies proper platform setup in the lab

### PR DESCRIPTION
**Pull request by JanPrusinowski**

### CODE REVIEW

File: dasharo-compatibility/basic-platform-setup.robot [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+BPS002.001 Boot to OS - Ubuntu
+    [Documentation]    This test verifies if platform can be booted to Ubunto and if correct credentials are set.
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+
+BPS002.002 Boot to OS - Windows
+    [Documentation]    This test verifies if platform can be booted to Windows, if SSH server is enabled and if correct credentials are set.
+    Power On
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:56]: Probably missing: `Boot System Or From Connected Disk` for Windows

File: dasharo-compatibility/basic-platform-setup.robot [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    ${out}=    Read From Terminal
+    Should Not Be Empty    ${out}
+
+BPS001.002 Power Control - Power Off
+    [Documentation]    This test verifies if the DUT can be powered down.
+    Power On
+    Sleep    60s
+    ${out}=    Read From Terminal
+    Should Not Be Empty    ${out}
+    Power Cycle Off
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:42]: Does it include testing Power button, or only RTE relay / sonoff? We need to ensure both buttons and power supply operate correctly.

File: dasharo-compatibility/basic-platform-setup.robot [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    Sleep    30s
+    ${out}=    Read From Terminal
+    Should Be Empty    ${out}
+
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:46]: Please add reset as well. `Rte Reset` should do the trick. For a full list of available kwds refer to: https://github.com/Dasharo/osfv-scripts/blob/main/osfv_cli/src/osfv/rf/rte_robot.py#L126

File: dasharo-compatibility/basic-platform-setup.robot [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    Login To Windows
+
+BPS003.001 External flashing
+    [Documentation]    This test verifies if flashrom can detect the die.
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    Get Flashrom From Cloud
+    ${out_flashrom}=    Execute Command In Terminal    flashrom -p internal
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:66]: This is internal flashing. You should use `Rte Flash Probe` instead: https://github.com/Dasharo/osfv-scripts/blob/main/osfv_cli/src/osfv/rf/rte_robot.py#L81

Comment (macpijan) [reply dasharo-compatibility/basic-platform-setup.robot:66]: I suggest following PASS conditions here: https://github.com/Dasharo/osfv-scripts/blob/main/osfv_cli/test/common/keywords.robot#L25

File: dasharo-compatibility/basic-platform-setup.robot [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    Should Contain    ${out_flashrom}    Found chipset
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:67]: This is for internal flashing.



I suggest following PASS conditions here: https://github.com/Dasharo/osfv-scripts/blob/main/osfv_cli/test/common/keywords.robot#L25

File: dasharo-compatibility/basic-platform-setup.robot (hunk 1 above)
Comment (JanPrusinowski) [reply dasharo-compatibility/basic-platform-setup.robot:56]: Actually its part of ```Login To Windows``` keyword

Comment (macpijan) [reply dasharo-compatibility/basic-platform-setup.robot:56]: right

File: dasharo-compatibility/basic-platform-setup.robot (hunk 4 above)
Comment (JanPrusinowski) [reply dasharo-compatibility/basic-platform-setup.robot:220]: Fixed in: 166e6685cb45534c1d0f3348b4d3f411f985bb60

File: dasharo-compatibility/basic-platform-setup.robot (hunk 5 above)
Comment (JanPrusinowski) [reply dasharo-compatibility/basic-platform-setup.robot:221]: Applied in: 166e6685cb45534c1d0f3348b4d3f411f985bb60

File: dasharo-compatibility/basic-platform-setup.robot (hunk 4 above)
Comment (macpijan) [reply dasharo-compatibility/basic-platform-setup.robot:220]: I do not see it being changed in the linked commit

File: dasharo-compatibility/basic-platform-setup.robot (hunk 5 above)
Comment (macpijan) [reply dasharo-compatibility/basic-platform-setup.robot:221]: as above

File: dasharo-compatibility/basic-platform-setup.robot [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+    Skip If    '${POWER_CTRL}' != 'RteCtrl'    DUT doesn't use RTE
+    Power On
+    Sleep    60s
+    FOR    ${i}    IN RANGE    30
+        ${out_temp}=    Read From Terminal
+        Sleep    10s
+        ${out}=    Read From Terminal
+        IF    '${out_temp}' == '${out}'    BREAK
+    END
+    Rte Reset    5
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:156]: Isn't pressing reset button for 5s rather long? Default 1s should be fine

File: dasharo-compatibility/basic-platform-setup.robot [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+    Login To Linux
+    Switch To Root User
+    Get Flashrom From Cloud
+    ${out_flashrom}=    Execute Command In Terminal    flashrom -p internal
+    Should Contain    ${out_flashrom}    Found chipset
+
+
+*** Keywords ***
+Check If Empty
+    [Documentation]    Checks if string is empty. Ignores spaces and newline.
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:226]: Please document return values

File: dasharo-compatibility/basic-platform-setup.robot [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+BPS001.001 Power Control - Power On and Serial output
+    [Documentation]    Verifies if the DUT can be turned On and if the serial output can be read.
+    Power On
+    FOR    ${i}    IN RANGE    120
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:33]: This seems to be repeated many times here, perhaps deservers a keyword?

File: dasharo-compatibility/basic-platform-setup.robot [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+        ${out}=    Read From Terminal
+        ${var}=    Check If Empty    ${out}
+        IF    ${var} == ${FALSE}    BREAK
+        Sleep    1s
+    END
+    IF    ${var}    Fail    Power On couldn't be detected
```
Comment (macpijan) [inline dasharo-compatibility/basic-platform-setup.robot:39]: Maybe more a matter of prerence, but https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Should%20Be%20True does the same , right?

File: dasharo-compatibility/basic-platform-setup.robot (hunk 5 above)
Comment (JanPrusinowski) [reply dasharo-compatibility/basic-platform-setup.robot:222]: Sorry. As above. I ment: 2ce08434ded82602cb43407e003a3469d9007a04

File: dasharo-compatibility/basic-platform-setup.robot (hunk 4 above)
Comment (JanPrusinowski) [reply dasharo-compatibility/basic-platform-setup.robot:221]: Sorry. I ment: 2ce08434ded82602cb43407e003a3469d9007a04

**Comment by macpijan** [comment]
Reworked and moved to util/

```
DEVICE_IP=192.168.10.136 FW_FILE=protectli_vp66xx_v0.9.0-rc6.rom RTE_IP=192.168.10.110 CONFIG=protectli-vp6670 ./scripts/run.sh  util/basic-platform-setup.robot
==============================================================================
Basic-Platform-Setup                                                          
==============================================================================
BPS001.001 Power Control - Power On and Serial output :: Verifies ... | PASS |
------------------------------------------------------------------------------
BPS001.002 Power Control - Power Off :: This test verifies if the ... | PASS |.
------------------------------------------------------------------------------
BPS002.001 RTE Relay low :: Verifies if RTE Relay set to low state... | SKIP |
DUT doesn't use RTE relay for power control
------------------------------------------------------------------------------
BPS002.002 RTE Relay high :: Verifies if RTE Relay set to high sta... | SKIP |
DUT doesn't use RTE relay for power control
------------------------------------------------------------------------------
BPS002.003 RTE Power On :: Verifies if Power Button can turn on th... | PASS |
------------------------------------------------------------------------------
BPS002.004 RTE Power Off :: Verifies if Power Button can turn off ... | PASS |
------------------------------------------------------------------------------
BPS002.005 RTE Reset :: Verifies if RTE Reset works                   | PASS |
------------------------------------------------------------------------------
BPS003.001 Sonoff Power On :: This test verifies if the DUT can be... | PASS |
------------------------------------------------------------------------------
BPS003.002 Sonoff Power Off :: This test verifies if the DUT can b... | PASS |
------------------------------------------------------------------------------
BPS004.001 Boot to OS - Ubuntu :: This test verifies if platform c... | PASS |
------------------------------------------------------------------------------
BPS004.002 Boot to OS - Windows :: This test verifies if platform ... | PASS |
------------------------------------------------------------------------------
BPS005.001 External flashing :: This test verifies if the flash di... | PASS |
------------------------------------------------------------------------------
BPS005.002 Internal flashing :: This test verifies if flashrom can... | PASS |
------------------------------------------------------------------------------
The asset has been checked out automatically. It is now automatically checked in as well. If you want to keep using it, reserve it manually first.
Basic-Platform-Setup                                                  | PASS |
13 tests, 11 passed, 0 failed, 2 skipped
==============================================================================
```

**Pull request merged**